src/client.rs

```diff
@@ -51,107 +57,97 @@ impl Client {
+    pub async fn get_tokio_connection(&self) -> RedisResult<crate::aio::Connection> {
+        use crate::aio::RedisRuntime;
+        Ok(
+            crate::aio::connect::<crate::aio::tokio::Tokio>(&self.connection_info)
+                .await?
+                .map(RedisRuntime::boxed),
+        )
     }
 
     /// Returns an async connection from the client.
     #[cfg(feature = "async-std-comp")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "tokio-comp")))]
     pub async fn get_async_std_connection(&self) -> RedisResult<crate::aio::Connection> {
-        crate::aio::connect_async_std(&self.connection_info).await
+        use crate::aio::RedisRuntime;
+        Ok(
+            crate::aio::connect::<crate::aio::async_std::AsyncStd>(&self.connection_info)
+                .await?
+                .map(RedisRuntime::boxed),
+        )
     }
 
     /// Returns an async connection from the client.
-    #[cfg(all(feature = "aio", feature = "tokio-rt-core"))]
+    #[cfg(any(feature = "tokio-rt-core", feature = "async-std-comp"))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(any(feature = "tokio-rt-core", feature = "async-std-comp")))
+    )]
     pub async fn get_multiplexed_async_connection(
         &self,
     ) -> RedisResult<crate::aio::MultiplexedConnection> {
-        #[cfg(all(feature = "tokio-comp", not(feature = "async-std-comp")))]
-        {
-            self.get_multiplexed_tokio_connection().await
-        }
-
-        #[cfg(all(not(feature = "tokio-comp"), feature = "async-std-comp"))]
-        {
-            self.get_multiplexed_async_std_connection().await
-        }
-
-        #[cfg(all(feature = "tokio-comp", feature = "async-std-comp"))]
-        {
-            if tokio::runtime::Handle::try_current().is_ok() {
-                self.get_multiplexed_tokio_connection().await
-            } else {
-                self.get_multiplexed_async_std_connection().await
-            }
-        }
-
-        #[cfg(all(not(feature = "tokio-comp"), not(feature = "async-std-comp")))]
-        {
-            compile_error!("tokio-comp or async-std-comp features required for aio feature")
+        match Runtime::locate() {
+            #[cfg(feature = "tokio-comp")]
+            Runtime::Tokio => self.get_multiplexed_tokio_connection().await,
+            #[cfg(feature = "async-std-comp")]
+            Runtime::AsyncStd => self.get_multiplexed_async_std_connection().await,
         }
     }
 
     /// Returns an async multiplexed connection from the client.
     ///
     /// A multiplexed connection can be cloned, allowing requests to be be sent concurrently
     /// on the same underlying connection (tcp/unix socket).
-    ///
-    /// This requires the `tokio-rt-core` feature as it uses the default tokio executor.
-    #[cfg(feature = "tokio-rt-core")]
     #[cfg(feature = "tokio-comp")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "tokio-rt-core")))]
+    #[cfg_attr(docsrs, doc(cfg(feature = "tokio-comp")))]
```

Comment:
Multiplexed connection needs spawn fn (provided by tokio/rt-core).

```suggestion
    #[cfg(feature = "tokio-rt-core")]
    #[cfg_attr(docsrs, doc(cfg(feature = "tokio-rt-core")))]
```